src/new/wallet.js

```diff
@@ -184,6 +184,129 @@ class HathorWallet extends EventEmitter {
+      const transaction = historyTransactions[i];
+
+      // Orphan transactions should be ignored
+      if (transaction.is_voided) continue;
+
+      // Iterate through outputs
+      for (let j = 0; j < transaction.outputs.length; j++) {
+        const output = transaction.outputs[j];
+
+        const is_unspent = output.spent_by === null;
+        if (!is_unspent) {
+          // No other checks required
+          continue;
+        }
+
+        const filters = wallet.filterUtxos(output, utxoDetails, options);
+        const is_authority = wallet.isAuthorityOutput(output);
+
+        // Max amount reached or max utxos.length, no more utxo should be added
+        if (!filters.is_max_amount_valid || !filters.is_max_utxos_valid) {
+          return utxoDetails;
+        }
+
+        if (filters.is_all_filters_valid && !is_authority) {
+          const locked = !wallet.canUseUnspentTx(output, j);
+          utxoDetails.utxos.push({
+            address: output.decoded.address,
+            amount: output.value,
+            tx_id: transaction.tx_id,
+            locked,
+            index: j,
+          });
+          if (!locked) {
+            utxoDetails.total_utxos_available++;
+            utxoDetails.total_amount_available += output.value;
+          } else {
+            utxoDetails.total_utxos_locked++;
+            utxoDetails.total_amount_locked += output.value;
+          }
+        }
+      }
+    }
+
+    return utxoDetails;
+  }
+
+  /**
+   * Consolidates many utxos into a single one
+   *
+   * @param {string} destinationAddress Address of the consolidated utxos
+   * @param {object} options Consolidation options (https://github.com/HathorNetwork/hathor-wallet-headless/issues/44)
+   *
+   * @return {object} { success: boolean, promise: Promise } indicates that the transaction is sent or not
+   */
+  async consolidateUtxos(destinationAddress, options = {}) {
+    storage.setStore(this.store);
+    const utxoDetails = this.getUtxos(options);
+    const inputs = [];
+    const utxos = [];
+    let total_amount = 0;
+    for (let i = 0; i < utxoDetails.utxos.length; i++) {
+      if (inputs.length === transaction.getMaxInputsConstant()) {
+        // Max number of inputs reached
+        break;
+      }
+      const utxo = utxoDetails.utxos[i];
+      if (utxo.locked) {
+        // Ignore locked utxo
+        continue
+      }
+      inputs.push({
+        hash: utxo.tx_id,
+        index: utxo.index,
+      });
+      utxos.push(utxo);
+      total_amount += utxo.amount;
+    }
+    const outputs = [{
+      address: destinationAddress,
+      value: total_amount,
+    }];
+    const token = {
+      uid: options.token || HATHOR_TOKEN_CONFIG.uid,
+      name: '',
+      symbol: ''
+    };
+
+    const result = this.sendManyOutputsTransaction(outputs, inputs, token);
+
+    if (!result.success) {
+      throw new Error(result.message);
+    }
+
+    const tx = await Promise.resolve(result.promise);
```

Comment:
Why do you await? I feel that the user should decide whether to await or not. So, I feel that we should return a promise here.

Reply:
`async/await` are just syntax sugar for Promises, so someone calling `consolidateUtxos` (which is an async method and already return a promise) can either do:
```js
const promise = hathorWallet.consolidateUtxos(...);
...
// Code that does not wait for utxo consolidation
...
promise.then(...).catch(...);
```
or
```js
const result = await hathorWallet.consolidateUtxos(...);
...
// Code that waits for utxo consolidation
```

If you recommend not proceed with `async/await` here, I can do something more similar to `sendManyOutputsTransaction` that returns a `Promise|{success: false, message: '...'}`. In my opinion, we should use `async/await` in asynchronous code whenever possible, mainly because it is easier to read.

Reply:
I agree that this syntax is way better to read and less error prone. Thanks!